src/Form/PathautoSettingsForm.php

```diff
@@ -183,7 +188,7 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
 
-    $config = $this->config('pathauto.settings');
+    $config = $this->configFactory()->getEditable('pathauto.settings');
```

Comment:
this shouldn't be necessary, config() uses the editable form if it's defined in getEditableConfigNames()